app/static/src/app/store/run/state.ts

```diff
@@ -16,19 +16,33 @@ export interface ParameterSet {
     hidden: boolean
 }
 
+export enum AdSettingCompType {
+    num = "numeric",
+    stdf = "standard form",
+    tag = "tag input"
+}
+
 type AdConfStandardForm = {
     val: [number|null, number|null],
     defaults: number[],
-    standardForm: true
+    type: AdSettingCompType.stdf
 }
 
 type AdConfNumeric = {
     val: number | null,
     defaults: number,
-    standardForm: false
+    type: AdSettingCompType.num
+}
+
+export type Tag = number | string
+
+type AdConfTCrit = {
+    val: Tag[] | null,
+    defaults: number[],
```

Comment:
Might the defaults include parameters not just numerics?
```suggestion
    defaults: Tag[],
```

Reply:
done

Reply:
wait, i just realised, defaults should be independent of parameters, like this default means the default for odin js and that cant have any knowledge of the parameters, i suspect we will always keep an empty array as default for this option anyway

Reply:
..oh so we're using the default as values that get fed directly into odin without going through the "tagsToOdin` or whatever it is? I can imagine people *might* want to default with parameters, but I guess don't need to support it initially.

Reply:
yep exactly, this default prop is more for the moment we feed into odin rather than anything with the UI, surely the way someone would set a default would be through their session info, like add tags and then load their session at some other point? and the advanced settings are stored in that, or do you mean set a default by configuring this in the wodin config at some point in the future?

Reply:
Yeah, I was imagining that AdvancedSettings defaults could eventually be part of the app config so include knowledge of parameters in the default code. But yeah, it's all hardcoded just now so fine for it to be empty numeric.